refgenconf/refgenconf.py

```diff
@@ -2087,10 +2162,134 @@ def _select_genomes(self, genome=None, strict=False, order=None,
                 else:
                     missing.append(g)
             if missing:
-                _LOGGER.warning("Genomes do not include: {}".format(", ".join(missing)))
+                _LOGGER.warning(
+                    "Genomes do not include: {}".format(", ".join(missing)))
             return None if not filtered else filtered
         return genomes if not all(x in genomes for x in genome) else genome
 
+    def config_upgrade(target_version, filepath, force=False):
+        """
+            upgrade the config file to a target version, 
+            and alter genome_folder stucture 
+
+            :param str target_version: the version updated to
+            :param str filepath: file path for the current config file
+            :param bool force
+        """
+
+        # check if any genome lack of local fasta asset and not on server
+
+        def check_genome_digests(config):
+            missing_digest = []
+            for k, v in config["genomes"].items():
```

Comment:
use the constant keys, defined in `refgenconf.const` module, instead of the literal ones, like `"genomes"` here.

Reply:
this applies to the entire code block

Reply:
bd369559d8354504a8f4a8b7769dd3e2e5f4661f